valhalla/midgard/openlr.h

```diff
@@ -160,24 +198,34 @@ struct LineLocation {
     attribute1 = raw[index++];
     auto attribute4 = raw[index++];
 
-    last = {longitude, latitude, attribute1, attribute4};
+    lrps.emplace_back(longitude, latitude, attribute1, attribute4);
 
     // Offsets
     std::bitset<8> flags(attribute4);
-    poff = (index < size) && flags[6] ? first.distance * raw[index++] / 256 : 0.f;
-    noff = (index < size) && flags[5] ? first.distance * raw[index++] / 256 : 0.f;
```

Comment:
this was a bug, the negative offset is from the end of the linear reference, so this would have only been correct for those linear references which were only 2 points (ie had no intermediate points)